core/client/src/main/java/alluxio/client/block/UnderStoreBlockInStream.java

```diff
@@ -156,22 +150,7 @@ public long skip(long n) throws IOException {
    * @param pos the position within this block
    * @throws IOException if the stream from the position cannot be created
    */
-  private void setUnderStoreStream(long pos) throws IOException {
-    if (mUnderStoreStream != null) {
-      mUnderStoreStream.close();
-    }
-    if (pos < 0 || pos > mLength) {
-      throw new IOException(ExceptionMessage.FAILED_SEEK.getMessage(pos));
-    }
-    UnderFileSystem ufs = UnderFileSystem.get(mUfsPath, ClientContext.getConf());
-    mUnderStoreStream = ufs.open(mUfsPath);
-    // The stream is at the beginning of the file, so skip to the correct absolute position.
-    if ((mInitPos + pos) != 0 && mInitPos + pos != mUnderStoreStream.skip(mInitPos + pos)) {
-      throw new IOException(ExceptionMessage.FAILED_SKIP.getMessage(pos));
-    }
-    // Set the current block position to the specified block position.
-    mPos = pos;
-  }
+  protected abstract void setUnderStoreStream(long pos) throws IOException;
```

Comment:
maybe I missed something but why we have `setUnderStoreStream` a part of abstract class?